src/image/PhotoCalib.cc

```diff
@@ -297,8 +297,9 @@ class PhotoCalibFactory : public table::io::PersistableFactory {
         table::BaseRecord const &record = catalogs.front().front();
         PhotoCalibSchema const &keys = PhotoCalibSchema::get();
         int version = getVersion(record);
-        if (version != 0) {
-            throw(pex::exceptions::TypeError("Unsupported version: " + std::to_string(version)));
+        if (version < 1) {
+            throw(pex::exceptions::RuntimeError("Unsupported version (version 0 was defined in maggies): " +
+                                                std::to_string(version)));
```

Comment:
Would there be any value to instead doing
```
auto mean = record.get(keys.calibrationMean);
auto err = record.get(keys.calibrationErr);
auto field = archive.get<afw::math::BoundedField>(record.get(keys.field));
if (version < 1) {
    field = (*field) * MAGGIES_TO_nJy;  // or whatever you've called it
    mean *= MAGGIES_TO_nJy;
    err *= MAGGIES_TO_nJy;
}
```

I think that question boils down to:
 - do `PhotoCalib`s in the wild actually implement `operator=`?
 - does this do the wrong thing anyway with version 0 `PhotoCalib`s in the wild that use meas_mosaic's `FluxFitBoundedField`?

Anyhow, if it *is* just as easy as the above, let's do the backwards compatibility.  But it's not worth lots of extra work if it isn't.

Reply:
The new version of `FluxFitBoundedField` has the conversion factor baked into it, so it would have to be special cased.

Reply:
I don't know the code well enough (read: at all), so I'm don't know how messy that special casing would be in practice — but if it's really just an `if` block with Jim's code in it, then it sounds like something we should do.

Reply:
`FluxFitBoundedField` is defined in meas_mosaic, which means we can't make a special case for it, now that I think about it (dependency goes the wrong way).

Reply:
My thought was that we could check for that type and only raise there, and convert what we can.

Reply:
The only thing I can think of that really matters is Dominique's pre-DC2 jointcal runs. Those are easy enough to redo.

Remember: the meas_mosaic PhotoCalib object is not used for any further processing that I'm aware of (e.g., the ffp object is still used to make coadds, even after DM-15751).

Reply:
There are no FGCM-produced PhotoCalibs that are out in the wild, and it would only take a few minutes to regenerate them anyway (since they're produced at the last step of processing). So if we can give a heads up to Dominique (or maybe update to slack somewhere) that might be useful (this would be before merger and the post-merger community post).